src/main/java/org/logstash/filters/geoip/Fields.java

```diff
@@ -96,6 +97,14 @@ public String getFieldReferenceECSv1() {
       Fields.COUNTRY_CODE3, Fields.IP, Fields.POSTAL_CODE, Fields.DMA_CODE, Fields.REGION_NAME,
       Fields.REGION_CODE, Fields.TIMEZONE, Fields.LOCATION, Fields.LATITUDE, Fields.LONGITUDE);
 
+  // When ECS is enabled, the composite REGION_ISO_CODE field is preferred to separate REGION_CODE
+  static final EnumSet<Fields> DEFAULT_ECS_CITY_FIELDS;
+  static {
+    DEFAULT_ECS_CITY_FIELDS = EnumSet.copyOf(DEFAULT_CITY_FIELDS);
+    DEFAULT_ECS_CITY_FIELDS.remove(REGION_CODE);
```

Comment:
This is perhaps a breaking change from a certain perspective, for a user who was relying on the non-ECS `[geo][region_code]` being populated in ECS mode. I'm inclined to call this a bugfix, since users have to explicitly opt into ECS to hit this behaviour.

But if we want to continue populating `[geo][region_code]` in ECS mode, this change and a small change to the specs would suffice:

```suggestion
```

Reply:
With ECS default on in 8.x, this is a breaking change. Users do not have the same data of `region_code` or replacement. Thinking to have both `region_code` and `region_iso_code`, but in some sense, the data of `region_code` is also a kind of iso code. Having both seems to be confusing. +1 to the breaking change.

Reply:
> With ECS default on in 8.x, this is a breaking change

A user upgrading from Logstash 7 to Logstash 8 should be expecting to be consuming breaking changes, and the path to ECS provides appropriate guidance on how to keep a plugin in non-ECS mode (hint: set `pipeline.ecs_compatibility: disabled` for the pipeline to lock in 7.x behaviour in _all_ its plugins).

It is _a little_ breaking for users who _already_ opt into ECS for the pipeline or an instance of this plugin, but it is also a bugfix since users who are already going out of their way to get ECS have a reasonable expectation that the result will be ECS (which `[geo][region_code]` is not).